beater/beater.go

```diff
@@ -75,7 +75,11 @@ func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 		logger:  logger,
 	}
 
-	if isElasticsearchOutput(b) && beaterConfig.Register.Ingest.Pipeline.isEnabled() {
+	// setup pipelines if explicitly directed to or setup --pipelines and config is not set at all
+	shouldSetupPipelines := beaterConfig.Register.Ingest.Pipeline.isEnabled() ||
+		(b.InSetupCmd && beaterConfig.Register.Ingest.Pipeline.Enabled == nil)
```

Comment:
no one will ever do `setup --pipelines -E register.ingest.pipeline.enabled=False`, but in that case behaviour here would be wrong, right?

Reply:
In that case `beaterConfig.Register.Ingest.Pipeline.Enabled` would not be `nil` and pipelines would not be loaded - would you expect them to be loaded no matter what instead?  There is a test (in python) for this case.

Reply:
right, never mind my boolean brain... :man_facepalming: